### ISSUE TITLE
error executing process while using empty cmd

### ISSUE THREAD

**Issue by RichardHCL** [spiffe/spiffe-helper#23]
While using **cmd = ""**, spiffe-helper keeps showing:
```
unable to signal process: error executing process: 
fork/exec : no such file or directory
```

< helper.conf with **cmd = ""** >
![Screen Shot 2020-05-12 at 12 01 19 PM](https://user-images.githubusercontent.com/9139322/81637476-0c69b880-9449-11ea-865a-35df6c14ba89.png)

< helper.conf with **cmd = "echo"** >
![Screen Shot 2020-05-12 at 12 01 51 PM](https://user-images.githubusercontent.com/9139322/81637479-0d9ae580-9449-11ea-8e4b-429c953a9c3c.png)

**Comment by marcosy** [comment]
Hi @RichardHCL 👋 
spiffe-helper currently requires the `cmd` configurable to be set. The provided command is invoked (or signaled if the process is running) every time certificates are rotated.

Curious to hear what your use case is for an empty `cmd`... would you expect it to only update the certificate files?

**Comment by RichardHCL** [comment]
Hi @marcosy ~~
Yes, I though it can update the certificate files without running any additional commands. And, I just got one more question, is there any specific command for cmd input or just general linux cmd would be sufficed?

**Comment by marcosy** [comment]
> I though it can update the certificate files without running any additional commands

Yes, that sounds reasonable to me. If `cmd` is unset or empty it should only update the certs.

>is there any specific command for cmd input or just general linux cmd would be sufficed?

Just general unix commands. If the value assigned to the `cmd` configurable does not contains path separators, spiffe-helper tries to resolve it to a complete path (like with the `echo` example). Otherwise it uses the `cmd` content as the path to an executable file (like `/Users/marcos/my-executable-file`).

**Comment by JU4N98** [comment]
You can assign me this issue.

**Comment by faisal-memon** [comment]
Thanks @JU4N98
